components/EmbedMedia/EmbedMedia.tsx

```diff
@@ -49,13 +50,48 @@ export const EmbedMedia = ({
    * https://github.com/cookpete/react-player/issues/1428
    */
   const [hasWindow, setHasWindow] = useState(false);
+  const playerWrapperRef = useRef<HTMLDivElement>(null);
+  const PreviewImg = previewImageSrc ? <PreviewImage previewImageSrc={previewImageSrc} /> : null;
+
   useEffect(() => {
-    if (typeof window !== 'undefined') {
-      setHasWindow(true);
-    }
+    setHasWindow(typeof window !== 'undefined');
   }, []);
 
-  const PreviewImg = previewImageSrc ? <PreviewImage previewImageSrc={previewImageSrc} /> : null;
+  // If the space key is pressed, do not scroll the page and click the element
+  const handleSpaceKeyDown = (event: KeyboardEvent) => {
+    if (event.code === 'Space' || event.key === ' ') {
+      event.preventDefault();
+      (event.target as HTMLElement).click();
+    }
+  };
+
+  useEffect(() => {
+    if (!isPreview || !playerWrapperRef.current) return;
+
+    const wrapper = playerWrapperRef.current;
+
+    /**
+     * Observe for an element with class .react-player__preview inside the wrapper
+     * and set its role to button when it is added to the DOM
+     */
+    const observer = new MutationObserver(() => {
+      const preview = wrapper.querySelector('.react-player__preview');
+      if (preview) {
```

Comment:
Do you need to check `if(preview)`? Does the element show up otherwise? Even if it does, do you care?